### PR TITLE
Users and Subscribers

### PR DESCRIPTION
**Pull request by Rostammahabadi**
### Type of change made:
- [x ] New feature (non-breaking change which adds functionality)

### Detailed Description
Created users table which have the following attributes: name, email, password, createdAt, updatedAt, thirty_days_from (will be used for blocking the user from going forward with a process that takes 30 days minimum to file/complete), ballotTitle, ballotDescription, meetUpAddress (for when they host meetupds)
Created mutations for users including userSignup (to create a user), userRemove( to delete a user), and updateUser which updates all fields for a user.
Added resolvers for the user mutations
Created Subscribers table which includes the following attributes:
email, createdAt, and updatedAt.
Created mutations to create and delete subscribers.
### Why is this change required? What problem does it solve?
Because we need it for the MVP
### Were there any challenges that arose while implementing this feature? If so, how were the addressed?
Many, and Google is my homie
### Files modified:
/api/src/config
/api/src/migrations
/api/modules
/api/seeders
/api/setup
database.js
graphql.js
load-modules.js
upload.js

Co-Authored-By: Danny Ramos <57972448+muydanny@users.noreply.github.com>
Co-Authored-By: Michelle <55991172+foleymichelle9@users.noreply.github.com>
Co-Authored-By: Mariana Cid <53917266+Mariana-21@users.noreply.github.com>

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  "main": "index.js",
+  "scripts": {
+    "setup": "npm run setup:install && npm run setup:db",
+    "setup:install": "npm install",
+    "setup:db": "npm run db:migrate && npm run db:seed",
+    "db:migrate": "./node_modules/.bin/sequelize db:migrate",
+    "db:seed": "./node_modules/.bin/sequelize db:seed:all",
+    "db:undo": "./node_modules/.bin/sequelize db:migrate:undo",
+    "db:seed:undo": "./node_modules/.bin/sequelize db:seed:undo",
+    "db:migrate:undo": "./node_modules/.bin/sequelize db:migrate:undo",
```
Comment (Rostammahabadi) [inline api/package.json:14]: Added these so you can run `npm run COMMANDS IN THE COMMENT`

File: api/src/migrations/1-user.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      meetUpAddress: {
+        allowNull: true,
+        type: Sequelize.STRING
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.dropTable('users');
+  }
+}
```
Comment (Rostammahabadi) [inline api/src/migrations/1-user.js:48]: migration to setup the user table, tested with seeders

File: api/src/migrations/2-subscribers.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.dropTable('subscribers');
+  }
+}
```
Comment (Rostammahabadi) [inline api/src/migrations/2-subscribers.js:26]: These will be for the users who want to get notified of when there is someone in the area getting signatures

File: api/src/modules/subscriber/mutations.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+export const subscriberRemove = {
+  type: SubscriberType,
+  args: {
+    id: {
+      name: 'id',
+      type: GraphQLInt
+    }
+  },
+  resolve: remove
+}
```
Comment (Rostammahabadi) [inline api/src/modules/subscriber/mutations.js:29]: We should only be adding and removing subscribers per MVP

File: api/src/modules/user/mutations.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      name: 'ballotTitle',
+      type: GraphQLString
+    },
+    ballotDescription: {
+      name: 'ballotDescription',
+      type: GraphQLString
+    }
+  },
+  resolve: updateUserResolver
+}
```
Comment (Rostammahabadi) [inline api/src/modules/user/mutations.js:74]: Updates the user based on what's passed in, if nothing's passed in then it doesn't get updated

File: api/src/modules/user/resolvers.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    const passwordMatch = await bcrypt.compare(password, userDetails.password)
+
+    if (!passwordMatch) {
+      // Incorrect password
+      throw new Error(`Sorry, the password you entered is incorrect. Please try again.`)
+    } else {
+      const userDetailsToken = {
+        id: userDetails.id,
+        name: userDetails.name,
+        email: userDetails.email,
```
Comment (Rostammahabadi) [inline api/src/modules/user/resolvers.js:48]: Kept this functionality due to having users login and logout

File: api/src/seeders/3-user.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        createdAt: new Date(),
+        updatedAt: new Date(),
+        meetUpAddress: "123 3rd street"
+      }
+    ])
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.bulkDelete('users', null, {});
+  }
+}
```
Comment (Rostammahabadi) [inline api/src/seeders/3-user.js:36]: Seeds to ensure the DB is setup correctly and data can be mutated/queried

File: api/package.json (hunk 1 above)
Comment (muydanny) [reply api/package.json:14]: oooo thats a script

File: api/src/migrations/1-user.js (hunk 2 above)
Comment (muydanny) [reply api/src/migrations/1-user.js:48]: meetUpAddress is a great addition

File: api/src/migrations/2-subscribers.js (hunk 3 above)
Comment (muydanny) [reply api/src/migrations/2-subscribers.js:26]: whats the reason behind the file being called 2-subscribers?

File: api/src/modules/subscriber/mutations.js (hunk 4 above)
Comment (muydanny) [reply api/src/modules/subscriber/mutations.js:29]: sick sick sick

**Review by muydanny — APPROVED** [review]

File: api/src/seeders/3-user.js (hunk 7 above)
Comment (muydanny) [reply api/src/seeders/3-user.js:36]: Ok, so were going to have users logging in?

Comment (Rostammahabadi) [reply api/src/seeders/3-user.js:36]: Yeah, just for the one's who want to go through the initiative process so we can save their initiative draft/meetupdate/ etc but not through GoogleOauth or anything just something internal

**Comment by cwefso** [comment]
Really incredible work on all of this. Great job, this is such a good foundation.

**Review by cwefso — APPROVED** [review]

**Review by Aeickelman40 — APPROVED** [review]
Great work Rostam! This was a major step forward in getting the BE and FE linked up.

File: api/src/migrations/2-subscribers.js (hunk 3 above)
Comment (Rostammahabadi) [reply api/src/migrations/2-subscribers.js:26]: Great question, Michelle and I ran into an issue where I undid the migration without undoing the seeds first so it got lost in limbo and GraphQL couldn't recognize that there was a new seed file to run so we had to change the number to 2.

File: api/src/migrations/1-user.js (hunk 2 above)
Comment (foleymichelle9) [reply api/src/migrations/1-user.js:48]: looks great!

**Review by foleymichelle9 — APPROVED** [review]
This all looks great, I like that we added 30 days from and meetUpAddress!

**Pull request merged**